src/CFNetwork/WebRequestStream.cs

```diff
@@ -74,18 +76,18 @@ public WebRequestStream (Stream stream, CancellationToken cancellationToken)
 
 		public async Task Open ()
 		{
-			writeStream.OpenCompletedEvent += (sender, e) => {
+			writeStream.OpenCompletedEvent += async (sender, e) => {
 				open = true;
-				openTcs.SetResult (null);
+				openTcs?.SetResult (null);
 				if (canWrite)
-					Write ();
+					await Write ();
 			};
-			writeStream.CanAcceptBytesEvent += (sender, e) => {
+			writeStream.CanAcceptBytesEvent += async (sender, e) => {
 				if (!open) {
 					canWrite = true;
 					return;
 				}
-				Write ();
+				await Write ();
```

Comment:
I really like the change

Reply:
I like it too, but it's also a behavior change in a PR with a lot of unrelated changes, so I think it might be best to have these changes in a different PR, especially since networking is code where we have limited test coverage.

Reply:
@rolfbjarne so would this look like just adding the ignore warning progma thing in this PR?

Reply:
@rolfbjarne Yes, setting `#nullable enable` enables 'Warnings as Errors' according to Sebastien's docs, so I cannot compile without setting ignores to any warnings / fixing them

Reply:
Ah ok, in that case yes, please add #pragma's to ignore the warnings